### PR TITLE
Automation test : Issue 47577: Add client API for renaming folders

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Create Endpoint test against java api story.

#### Related Pull Requests
- https://github.com/LabKey/platform/issues/4323
- https://github.com/LabKey/platform/pull/4400

#### Changes
*

### CODE REVIEW

File: src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        log("Required values missing: Verify error appears");
+        command = new RenameContainerCommand(null, null, false);
+        try
+        {
+            command.execute(cn, getProjectName());
+        }
+        catch (CommandException e)
+        {
+            Assert.assertEquals("Blank names are not allowed.", 500, e.getStatusCode());
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java]: @RosalineP This seems wrong. Bad client input should be some 4xx error (400 or 422 maybe?). 500 means the server did something wrong.

File: src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        Assert.assertEquals("Rename container api failed ", 200, response.getStatusCode());
+        Assert.assertEquals("Incorrect new name", newContainerName + "1", response.getName());
+        Assert.assertEquals("Incorrect new title", newTitle + "1", response.getProperty("title"));
+
+        log("Required values missing: Verify error appears");
+        command = new RenameContainerCommand(null, null, false);
+        try
+        {
+            command.execute(cn, getProjectName());
+        }
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java]: If you're expecting an error, this should fail when an error doesn't occur.
```suggestion
        try
        {
            command.execute(cn, getProjectName());
            Assert.fail("Rename should have failed");
        }
```

File: src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java (hunk 1 above)
Comment (RosalineP) [reply src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java]: Oh shoot, I was using ApiUsageExceptions, which do 500 out. I'll fix this server-side
Update: New branch made

**Review by RosalineP — APPROVED** [review]
Coverage looks good!

**Pull request merged**